packages/apollo-link-accounts/src/index.ts

```diff
@@ -5,14 +5,19 @@ import { AccountsClient } from '@accounts/client';
 type AccountsClientFactory = () => AccountsClient | Promise<AccountsClient>;
 
 export const accountsLink = (accountsClientFactory: AccountsClientFactory): ApolloLink => {
-  return setContext(async (_, { headers: headersWithoutTokens }) => {
+  return setContext(async (req, { headers: headersWithoutTokens }) => {
     const accountsClient = await accountsClientFactory();
-    const tokens = await accountsClient.refreshSession();
-
     const headers = { ...headersWithoutTokens };
 
-    if (tokens) {
-      headers.Authorization = 'Bearer ' + tokens.accessToken;
+    /**
+     * We have to check this condition to avoid an infinite loop
+     * during the refresh token operation
+     */
+    if (req.operationName !== 'refreshTokens') {
```

Comment:
@pradel isn't 'refreshTokens' a string constant? is there anyway we can use that constant instead of using a raw value here?

Reply:
(I know you merge the PR, but I thought I should point out this one)

Reply:
@thomasttvo that's a good idea since it's a constant, but I am not sure how to achieve that in the current setup 😁